akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberShards.scala

```diff
@@ -16,33 +21,60 @@ import akka.persistence.PersistentActor
 private[akka] object EventSourcedRememberShards {
   def props(typeName: String): Props =
     Props(new EventSourcedRememberShards(typeName))
+
+  class FromOldCoordinatorState() extends EventAdapter {
+    override def manifest(event: Any): String =
+      ""
+
+    override def toJournal(event: Any): Any =
+      event
+
+    override def fromJournal(event: Any, manifest: String): EventSeq = {
+      event match {
+        case ShardHomeAllocated(shardId, _) =>
+          EventSeq.single(shardId)
+        case _ => EventSeq.empty
+      }
+
+    }
+  }
 }
 
 /**
  * INTERNAL API
+ *
+ * FIXME add snapshotting and then the migration event adapter can be removed
  */
 @InternalApi
-private[akka] final class EventSourcedRememberShards(typeName: String) extends PersistentActor {
+private[akka] final class EventSourcedRememberShards(typeName: String) extends PersistentActor with ActorLogging {
 
-  override val persistenceId: String = s"$typeName-remember-entitites"
+  // Uses the same persistence id as the old persistent coordinator so that the old data can be migrated
+  // without any user action
+  override def persistenceId = s"/sharding/${typeName}Coordinator"
```

Comment:
One thing I thought about today was that we should pick up journal and snapshot store from ClusterShardingSettings rather than always use default

Reply:
👍